docs/source/reference/types.rst

```diff
@@ -0,0 +1,286 @@
+This is desugared to ``{0 x : _} -> a``.
+
+^^^^^^^^^^^^^
+Implicits
+^^^^^^^^^^^^^
+
+Implicit function types, those where the value of a parameter can be inferred from the context, can take three forms.
+The first of these, ``{x : A} -> B``, forms the basis "basic" implicit types.
+That is, ``x`` should be inferred *syntactically*.
+This is the equivalent of implicit ``forall`` in Haskell or Rocq, where we might state that ``vectorMap`` on a vector might be:
+
+.. code:: idris
+
+    vectorMap : {n : Nat} -> (a -> b) -> Vect n a -> Vect n b
+
+We would then use ``vectorMap`` as follows:
+
+.. code-block:: idris
+
+    vectorMap f v
+
+or
+
+.. code:: idris
+
+    vectorMap {n = m} f v
+
+These two examples show the power of Idris implicits: while they can be inferred by the compiler, if necessary (or desired) they may be specified by the user
+
+Idris has *two* additional modifiers of implicits, however.
+The first, while not that important to the type system of Idris, is *incredibly* useful, and it is the default implicit.
+Essentially "regular" implicits only try to figure out the type without adding any extra information.
+
+Default implicits, on the other hand, instead of trying to figure out a value, will instead just use a default, which means that unnecessary "simple" functions are unnecessary in Idris: you just use the default value.
+
+The syntax for default implicits is ``{default val x : a} -> b``, where ``val : a``, which is the default value for ``x``.
+
+The third kind of implicits are auto implicits.
+Auto implicits are most useful in proofs, where they differ from regular implicits by there purpose.
+Regular implicits try to find a "necessary" match.
+``auto`` implicits, on the other hand, look for *any* possible construction of a value.
+This is useful for propositions, where we often don't need to reason about the specific value of a type member, merely that it exists.
+Essentially, auto implicits are for cases where we want the compiler to have more freedom to infer the value
+
+For instance, if we want to automatically pass a proof through some functions, we use ``auto`` defaults.
+
+.. hint::
+    Lowercase names in types are automatically universally quantified in types.
+    For instance, ``map : (a -> b) -> List a -> List b`` is transformed into ``map : {0 a : Type} -> {0 b : Type} -> (a -> b) -> List a -> List b`` [#f1]_.
+    You can write this signature explicitly, and it will compile the same as the simple version, and in addition you could also have the ``forall`` keyword and wrote this as ``map : forall a . forall b . ((a -> b) -> List a -> List b)``.
+
+    If desired, this behavior can be toggled off by ``%unbound_implicits off`` (and back on by ``%unbound_implicits on``)
+
+
+
+^^^^^^^^^^^^^
+Quantities
+^^^^^^^^^^^^^
+
+Idris 2 uses Quantitative Type Theory (`QTT`_) as its theoretical basis.
+In QTT, every function type has an associated multiplicity, ``0``, ``1``, or ω.
```

Comment:
```suggestion
In QTT, every variable has an associated multiplicity, ``0``, ``1``, or ω.
```

Essentially every binder  `a : b` has a quantity attached to them: lambdas, pi-types, record fields, toplevel definitions, etc. probably important to explain that in some way. "Variable" is probably good enough for an entry-level document